openbb_platform/CONTRIBUTING.md

```diff
@@ -1,65 +1,179 @@
+    volume: float = Field(description=DATA_DESCRIPTIONS.get("volume", ""))
+    vwap: Optional[PositiveFloat] = Field(description=DATA_DESCRIPTIONS.get("vwap", ""), default=None)
+```
+
+The `Data` class is an abstract class that tells us the expected output data. Here we can see a `vwap` field that is `Optional`. This is because not all providers share this field while it is shared between two or more providers.
+
+## What is an extension?
+
+An extension adds functionality to the OpenBB Platform. It can be a new data source, a new command, a new visualization, etc.
+
+### Types of extensions
+
+We primarily have 3 types of extensions:
+
+1. OpenBB Extensions - built and maintained by the OpenBB team (e.g. `openbb-stocks`)
+2. Community Extensions - built by anyone and primarily maintained by OpenBB (e.g. `openbb-yfinance`)
+3. Independent Extensions - built and maintained independently by anyone
+
+If your extension is of high quality and you think that it would be a good community extension, you can open a PR to the OpenBB Platform repository and we'll review it.
+
+We encourage independent extensions to be shared with the community by publishing them to PyPI.
+
+# How to build OpenBB extensions?
+
+We have a Cookiecutter template that will help you get started. It serves as a jumpstart for your extension development, so you can focus on the data and not on the boilerplate.
 
 Please refer to the [Cookiecutter template](https://github.com/OpenBB-finance/openbb-cookiecutter) and follow the instructions there.
 
-This document will walk you through the steps of adding a new custom extension to the Platform. The high level steps are:
+This document will walk you through the steps of adding a new extension to the OpenBB Platform.
+
+The high level steps are:
 
 - Generate the extension structure
 - Install your dependencies
 - Install your new package
 - Use your extension (either from Python or the API interface)
+- QA your extension
+- Share your extension with the community
 
-## Cookiecuter, a closer look
-
-The Cookiecutter template generates a set of files in which we can find instructions and explanations.
+## Add a custom data source
 
-> Note that the code is functional, so you can just run it and start playing with it.
+You will get your data either from a CSV file, local database or from an API endpoint.
 
-### Get your data
-
-Either from a CSV file, local database or from an API endpoint, you'll need to get your data.
-
-If you don't want to partake in the data standardization framework, you can simply write all the logic straight inside the router file. This is usually the case when you are adding unique data that isn't easily standardizable.
+If you don't want to partake in the data standardization framework, you can simply write all the logic straight inside the router file. This is usually the case when you are adding alternative data that isn't easily standardizable.
```

Comment:
How does this work with the unit testing? There is an expected, rigid, formatting throughout, both the provider-level and the router-level.

Reply:
In this case, where you defined everything inside the router file, you won't have a Fetcher at all. We don't yet have unit tests for commands themselves but test the Fetchers. This will also need to be handled on the integration test side soon. @hjoaquim and I and working on this.

Reply:
We also need to provide more clarity on the distinct differences between the Fetcher and the Router and the rigid output requirements of each.  Fetcher does not replace Command Runner, they are both required.  If we don't provide these specific caveats, people are going to just send a DataFrame to the output and not know why it doesn't work.